server/src/main/java/io/crate/metadata/settings/session/SessionSettingRegistry.java

```diff
@@ -155,10 +173,42 @@ public Map<String, SessionSetting<?>> settings() {
     }
 
     private static String[] objectsToStringArray(Object[] objects) {
-        String[] strings = new String[objects.length];
+        ArrayList<String> argumentList = new ArrayList<>();
         for (int i = 0; i < objects.length; i++) {
-            strings[i] = DataTypes.STRING.implicitCast(objects[i]);
+            String str = DataTypes.STRING.implicitCast(objects[i]).trim();
+            argumentList.addAll(Arrays.stream(str.split(","))
+                              .map(String::trim)
+                              .collect(Collectors
+                              .toCollection(ArrayList::new)));
+        }
+        return argumentList.toArray(String[]::new);
```

Comment:
@matriv could you maybe also take a look here.

This also affects `SET search_path` and the current implementation allows to set `search_path`s with schemas leading/trailing whitespaces (which are supported by PG to some degree, but not by CrateDB). 

e.g. (5.0)
```sql
cr> SET search_path TO 'nasty , pg_catalog  ', nots;                                                              
SET OK, 0 rows affected  (0.000 sec)
cr> show search_path;                                                                                             
+----------------------------------------+
| setting                                |
+----------------------------------------+
| pg_catalog, nasty , pg_catalog  , nots |
+----------------------------------------+
SHOW 1 row in set (0.002 sec)
```

which also circumvents checking for `pg_catalog` in the search_space and automatically adding it at the beginning.

with this change:
```sql
cr> SET search_path TO 'nasty , pg_catalog  ', nots;                                                              
SET OK, 0 rows affected  (0.010 sec)
cr> show search_path;                                                                                             
+-------------------------+
| setting                 |
+-------------------------+
| nasty, pg_catalog, nots |
+-------------------------+
SHOW 1 row in set (0.002 sec)
```

---

also open for feedback if this should be separated out.
No tests are affected by this change.

Reply:
Seems a nice "side-effect" to me, let's just add a test to demonstrate this.

Reply:
added a test for `search_path`